src/debug/index.ts

```diff
@@ -61,4 +62,41 @@ export abstract class MesonDebugConfigurationProvider implements vscode.DebugCon
   ): vscode.ProviderResult<vscode.DebugConfiguration> {
     return debugConfiguration;
   }
+
+  async createDebugConfigurationForType(target: Target, type: DebuggerType): Promise<vscode.DebugConfiguration> {
+    const targetName = await getTargetName(target);
+    const name = type.toString();
+    let debugConfig: vscode.DebugConfiguration = {
+      type: name,
+      name: `Debug ${target.name} (${name})`,
+      request: "launch",
+      cwd: path.dirname(this.path),
+      program: target.filename[0],
+      args: [],
+      preLaunchTask: `Meson: Build ${targetName}`,
+    };
+
+    if (type === DebuggerType.cppdbg) {
+      let miMode;
+      if (
+        target.target_sources?.some((source) => source.compiler != null && ["cc", "clang"].includes(source.compiler[0]))
+      ) {
```

Comment:
Not sure I understand why cc means clang.

Reply:
This is definitively a wrong assumption. Since it's hard to infer it, I suggest using the [defaults](https://code.visualstudio.com/docs/cpp/launch-json-reference#_customizing-gdb-or-lldb) based on the operating system. This could be customizable in the future with a new configuration property.